apisix/plugins/error-log-logger.lua

```diff
@@ -272,6 +333,52 @@ local function send_to_clickhouse(log_message)
 end
 
 
+local function create_producer(broker_list, broker_config, cluster_name)
+    core.log.info("create new kafka producer instance")
+    return producer:new(broker_list, broker_config, cluster_name)
+end
+
+
+local function send_to_kafka(log_message)
+    core.log.info("sending a batch logs to kafka brokers: ",
+                   core.json.delay_encode(config.kafka.brokers))
+
+    local broker_config = {}
+    broker_config["request_timeout"] = config.timeout * 1000
+    broker_config["producer_type"] = config.kafka.producer_type
+    broker_config["required_acks"] = config.kafka.required_acks
+
+    local metadata = plugin.plugin_metadata(plugin_name)
```

Comment:
It seems using the config passed from `process` here might create a race:

Consider we have c1(config) and m1 (modifiedIndex) in `process`, and c2/m2 in `send`. It looks like we might use m2 as key and c1 as value in the cache below.

Reply:
>  It looks like we might use m2 as key and c1 as value in the cache below.

`process` and `send` only read data from plugin_metadata, this case maybe not occur?

Reply:
> It looks like we might use m2 as key and c1 as value in the cache below.

if you mean that we need to clone `config` before we use it like below? then we can use m2 and c2 in `send`.

`local conf = core.table.clone(config)`

Reply:
A clone of c1 still has a c1's value.
Maybe we can get the c2 in `send` like what we have done in `process`.

Reply:
ok, i will change the code and update the PR later.

Reply:
@spacewander Could you please take a look, thanks.